src/simularium/rendering/InstancedFiberShader.ts

```diff
@@ -262,9 +262,42 @@ void createTube (float t, vec2 volume, out vec3 offset, out vec3 normal) {
   vec3 prev = sampleCurve(t1);
   vec3 next = sampleCurve(t2);
 
-  // compute the TBN matrix
+  // compute the TBN matrix (aka the Frenet-Serret frame)
+
+  // tangent is just the direction along our small delta
   vec3 T = normalize(next - prev);
-  vec3 B = normalize(cross(T, next + prev));
+
+  vec3 B = vec3(0, 0, 0);
+
+  // if next-prev and next+prev are parallel, then
+  // our normal and binormal will be ill-defined so we need to check for that
+  // check parallel by dot the unit vectors and check close to +/-1
+  float check = dot(T, normalize(next + prev));
+  if (abs(check) < 0.999) {
+    // cross product of parallel vectors is 0, so T must not be parallel to next+prev
+    // hence the above check.
+    B = normalize(cross(T, next + prev));
+  }
+  else {
+    // special case for which N and B are not well defined. 
+    // so we will just pick something
```

Comment:
I think it would be helpful to put here "so we will just pick an arbitrary vector that's not parallel to T" (if I remembered that correctly?). Actually, how does this algorithm ensure finding a vector that's not parallel to T?

Reply:
It's not this exact algorithm but it's something similar to these: 
https://math.stackexchange.com/questions/3122010/how-to-deterministically-pick-a-vector-that-is-guaranteed-to-be-non-parallel-to
I actually feel a bit more comfortable with those answers so I might change this at some point.